src/Umbraco.Web.UI.Client/src/common/directives/components/umbdatetimepicker.directive.js

```diff
@@ -0,0 +1,143 @@
+/**
+@ngdoc directive
+@name umbraco.directives.directive:umbDateTimePicker
+@restrict E
+@scope
+
+@description
+Use this directive to render a date time picker
+
+<h3>Markup example</h3>
+<pre>
+	<div ng-controller="My.Controller as vm">
+
+        <umb-date-time-picker
+            options="vm.config"
+            ng-model="vm.date"
+            on-change="vm.datePickerChange(event)"
+            on-error="vm.datePickerError(event)">
+        </umb-date-time-picker>
+
+	</div>
+</pre>
+
+<h3>Controller example</h3>
+<pre>
+	(function () {
+		"use strict";
+
+		function Controller() {
+
+            var vm = this;
+
+            vm.date = "";
+
+            vm.config = {
+                pickDate: true,
+                pickTime: true,
+                useSeconds: true,
+                format: "YYYY-MM-DD HH:mm:ss",
+                icons: {
+                    time: "icon-time",
+                    date: "icon-calendar",
+                    up: "icon-chevron-up",
+                    down: "icon-chevron-down"
+                }
+            };
+
+            vm.datePickerChange = datePickerChange;
+            vm.datePickerError = datePickerError;
+
+            function datePickerChange(event) {
+                // handle change
+            }
+
+            function datePickerError(event) {
+                // handle error
+            }
+
+        }
+
+		angular.module("umbraco").controller("My.Controller", Controller);
+
+	})();
+</pre>
+
+@param {object} options (<code>binding</code>): Config object for the date picker.
+@param {string} ngModel (<code>binding</code>): Date value.
+@param {callback} onChange (<code>callback</code>): Change callback.
+@param {callback} onError (<code>callback</code>): Error callback.
+**/
+
+(function () {
+    'use strict';
+
+    function DateTimePickerDirective(assetsService) {
+
+        function link(scope, element, attrs, ctrl) {
+
+            function onInit() {
+                // load css file for the date picker
+                assetsService.loadCss('lib/datetimepicker/bootstrap-datetimepicker.min.css').then(function () {
+
+                });
+                
+                // load the js file for the date picker
+                assetsService.loadJs('lib/datetimepicker/bootstrap-datetimepicker.js').then(function () {
+                    // init date picker
+                    initDatePicker();
+                });
+            }
+
+            function onChange(event) {
+                if (scope.onChange && event.date && event.date.isValid()) {
+                    scope.$apply(function(){
+                        // Update ngModel
+                        scope.ngModel = event.date.format(scope.options.format);
```

Comment:
Is the format used for display purposes or is the format used when being sent back to the server endpoints, as to make life easier with datetimes, formats & timezone hell, I think it may need to be consistently sent to the server in a format that is agreed upon, as if you were to change it clientside the server side may interpret the date as the totally wrong month.